Unexpected error using balanced option in spaced-comment rule

**What version of ESLint are you using?**

3.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

```yaml
rules:
  spaced-comment:
    - error
    - always
    - block:
        balanced: true
```

**What did you do? Please include the actual source code causing the issue.**

Linted this file:

```js
/* this line is fine */

/*
 * these lines are fine
 */

// this line is reported as a problem
```

**What did you expect to happen?**

- No linting errors to be detected.
- The `balanced` option to only be considered when analyzing `/* */` style comments, not `//` style.

**What actually happened? Please include the actual, raw output from ESLint.**

Encountered this linting error:

```
7:1  error  Expected space or tab before '*/' in comment  spaced-comment

✖ 1 problem (1 error, 0 warnings)
```

Looks like changing the file so it only contains one line results in the same error:

File:

```js
// this line is reported as a problem
```

Result:

```
  1:1  error  Expected space or tab before '*/' in comment  spaced-comment

✖ 1 problem (1 error, 0 warnings)

```

--- a/lib/rules/spaced-comment.js
+++ b/lib/rules/spaced-comment.js
@@ -342,7 +342,7 @@ module.exports = {
                     }
                 }
 
-                if (balanced && !endMatch) {
+                if (balanced && type === "block" && !endMatch) {
                     reportEnd(node, "Expected space or tab before '*/' in comment");
                 }
             } else {
@@ -354,7 +354,7 @@ module.exports = {
                     }
                 }
 
-                if (endMatch && balanced) {
+                if (balanced && type === "block" && endMatch) {
                     reportEnd(node, "Unexpected space or tab before '*/' in comment", endMatch);
                 }
             }

--- a/tests/lib/rules/spaced-comment.js
+++ b/tests/lib/rules/spaced-comment.js
@@ -171,6 +171,14 @@ ruleTester.run("spaced-comment", rule, {
             code: "//\n",
             options: ["always"]
         },
+        {
+            code: "// space only at start; valid since balanced doesn't apply to line comments",
+            options: ["always", { block: { balanced: true }}]
+        },
+        {
+            code: "//space only at end; valid since balanced doesn't apply to line comments ",
+            options: ["never", { block: { balanced: true }}]
+        },
 
         // block comments
         {

